### ISSUE TITLE
Conversion/Move to C# .NET/MONO

### ISSUE THREAD

**Issue by thecoder75** [thecoder75/liveme-pro-tools#32]
From @monstergarden:

I will do a first test soon. Goals:

- the electron app will stay the same, but we can slowly migrate parts to it to c#, making it more robust and more enjoyable to write (intellisense)
- electron process will on startup start the (dotnet core asp.net) background process.
- create a helper class for interacting with the background process
- create build pipeline and verify it works for developers on all plattforms
- verify that the resulting executable will work on all users machines, without the need to install something.
- write documentation for getting started (for developers)

**Comment by monstergarden** [comment]
Yesterday I evaluated server-side blazor (razor components) for using c#/backend for DOM manipulation.

This has the advantage that we wouldn't need to create a REST API for all backend functionalities ( which can be a tideous task), but instead just call the functions / classes directly within the UI logic. This would also allow to update the UI while a process is ongoing - compared to a REST Api which provides no functionality to get intermediate results.

https://youtu.be/GI_9g9lZpik?t=2290

Blazor also has a Javascript interop. So we can still go with the iterative migration approach.
As a first test I rewrote the bookmarks feed on the homescreen, but was using the exisiting JS classes like liveme.js and datamanger.js to get the data but all the UI logic was written in C#.



The only downside is, that the UI is not rendering, if the backend is not running.
For better UX we should probably create a separate start screen which is not rendered in the backend, which should allow troubleshooting and give feedback to the user in case of errors (like a button to restart the backend process).

I will further testings / cleanup before pushing the results.
